ProfileManager.js

```diff
@@ -140,10 +149,10 @@ export default class ProfileManager {
     if(agentZcap) {
       return agentZcap;
     }
-
     const promise = this._getAgentCapability(
       {id, profileAgent, useEphemeralSigner});
-    capabilityCache.set(capabilityKey, promise);
+    const maxAge = this.zcapTtl - this.zcapGracePeriod;
```

Comment:
We need to use the `expires` field from the zcap here because it may not match this calculation. So it should be when the zcap expires - the grace period. Same as below with the agent signer, I'd imagine.